foundationdb/src/tuple/element.rs

```diff
@@ -273,29 +275,38 @@ impl Encode for Vec<Element> {
                 }
             }
         }
-        NIL.write(w)
+
+        if tuple_depth.depth() > 0 {
+            NIL.write(w)?;
+        }
+
+        Ok(())
     }
 }
 
 impl Decode for Vec<Element> {
     fn decode(mut buf: &[u8], tuple_depth: TupleDepth) -> Result<(Self, usize)> {
-        if buf.len() < 2 {
+        if tuple_depth.depth() > 0 && buf.len() < 2 {
             return Err(Error::EOF);
         }
 
-        // TODO: should this only write the Nested markers in the case of tuple_depth?
-        NESTED.expect(buf[0])?;
         let len = buf.len();
-        buf = &buf[1..];
+        let nested = buf.len() > 0 && buf[0] == NESTED;
+        if nested {
+            buf = &buf[1..];
+        }
 
         let mut tuples = Vec::new();
         loop {
             if buf.is_empty() {
-                // tuple must end with NIL byte
-                return Err(Error::EOF);
+                if nested {
```

Comment:
It will never enter here if it's empty, it will just silently break.

Reply:
yeah